debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCArrayletObjectModelBase.java

```diff
@@ -242,20 +259,28 @@ protected long getArrayLayout(J9ArrayClassPointer clazz, UDATA dataSizeInBytes)
 			}
 		} else {
 			UDATA lastArrayletBytes = dataSizeInBytes.bitAnd(arrayletLeafSizeMask);
+			boolean isAllIndexableDataContiguousEnabled = enableVirtualLargeObjectHeap || enableDoubleMapping;
+
+			if (isAllIndexableDataContiguousEnabled && dataSizeInBytes.gt(0)) {
+				layout = GC_ArrayletObjectModelBase$ArrayLayout.InlineContiguous;
+			} else if (lastArrayletBytes.gt(0)) {
+				/* determine how large the spine would be if this were a hybrid array */
+				UDATA numberArraylets = numArraylets(dataSizeInBytes);
+				boolean align = shouldAlignSpineDataSection(clazz);
+				UDATA hybridSpineBytes = getSpineSize(GC_ArrayletObjectModelBase$ArrayLayout.Hybrid, numberArraylets, dataSizeInBytes, align);
+				UDATA adjustedHybridSpineBytes = ObjectModel.adjustSizeInBytes(hybridSpineBytes);
+				UDATA adjustedHybridSpineBytesAfterMove = adjustedHybridSpineBytes;
+				if (GCExtensions.isVLHGC()) {
+					adjustedHybridSpineBytesAfterMove.add(ObjectModel.getObjectAlignmentInBytes());
```

Comment:
what is this doing - adding size of dataAddr field that is 8, which coincide with object aligment in most cases (except 4-bit shift CR)
if so:
- try to find a better check for 'does dataAddr exists' instead of isVLHGC
- try to find a better size (size of UDATA etc) so it works for 4bit CR